src/Ast/ExprResolver.dfy

```diff
@@ -118,6 +119,9 @@ module ExprResolver {
         var trs :- ResolvePatterns(patterns, ers, varMap');
         var _ :- CheckPatterns(trs, boundVars);
         r := QuantifierExpr(univ, boundVars, trs, b);
+      case ClosureExpr(closureBindings, resultVar, resultType, properties) =>
+        var elaborated :- ElaborateClosure(expr, ers, varMap);
+        r := elaborated;
```

Comment:
You don't need to introduced the variable `elaborated`. Instead, you can assign directly to `r`, like

    r :- ElaborateClosure(expr, ers, varMap);